### PR TITLE
Assorted cleanup of CI/packaging related code.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

- Remove old Dockerfile.test. It hasn’t been used for a long time, and doesn’t even do what it was intended for at this point.
- Move the Coverity scan script to `packaging/utils`.
- Remove CodeClimate configuration. We’re not using it anywhere.
- Remove cppcheck script and config, as they are also not really used anymore (and the config does not appear to be up to date anyway).
- Remove the `build_external` directory, as the stuff in it appears to be completely unused, and even if it _is_ used it should be located elsewhere in the repo (probably under `tests`).
- Remove `.gitattributes` since it seems to be unused.
- Remove the devcontainer config as it seems to be unused.
- Move extra CMake related stuff under `packaging/cmake`.

##### Test Plan

CI passes on this PR.

### CODE REVIEW

File: .github/scripts/coverity-scan.sh [hunk 1]
```diff
@@ -42,7 +42,18 @@ INSTALL_DIR="/opt"
 # the version of coverity to use
 COVERITY_BUILD_VERSION="${COVERITY_BUILD_VERSION:-cov-analysis-linux64-2023.6.2}"
 
-. packaging/installer/functions.sh
+SCRIPT_SOURCE="$(
```
Comment (tkatsoulas) [inline .github/scripts/coverity-scan.sh:45]: Could you please rename to something more insidefull like CALLING_SCRIPT_PATH or INVOKER_SCRIPT_PATH

```suggestion
CALLING_SCRIPT_PATH="$(
```

Comment (Ferroin) [reply .github/scripts/coverity-scan.sh:45]: It’s the location of the script itself, not anything about how it was called or invoked.

File: packaging/utils/coverity-scan.sh [hunk 2]
```diff
@@ -42,7 +42,19 @@ INSTALL_DIR="/opt"
 # the version of coverity to use
 COVERITY_BUILD_VERSION="${COVERITY_BUILD_VERSION:-cov-analysis-linux64-2023.6.2}"
 
-. packaging/installer/functions.sh
+SCRIPT_SOURCE="$(
```
[the remark this answers is not in the document]
Comment (Ferroin) [reply packaging/utils/coverity-scan.sh:45]: The script isn’t just run in CI, and I would rather try to ensure it doesn’t depend on where exactly it’s run from (I’ve been bitten by such assumptions too many times in the past).

[the remark this answers is not in the document]
Comment (vkalintiris) [reply packaging/utils/coverity-scan.sh:45]: More code -> more stuff to maintain and understand. Not my style.

**Review by vkalintiris — APPROVED** [review]

**Pull request merged**